Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -57,9 +59,135 @@ namespace AzAssetBrowserRequestHandlerPrivate
 {
     using namespace AzToolsFramework;
     using namespace AzToolsFramework::AssetBrowser;
-    // return true ONLY if we can handle the drop request in the viewport.
-    bool CanSpawnEntityForProduct(const ProductAssetBrowserEntry* product,
-        AZStd::optional<const AZStd::vector<AZ::Data::AssetType>> optionalProductAssetTypes = AZStd::nullopt)
+
+    bool ProductHasAssociatedComponent(const ProductAssetBrowserEntry* product);
+
+    // given a list of products all belonging to the same parent source file
+    // select just one that can best represent the entire source file.
+    // it does so by collecting all of the spawnable assets and sorts them using
```

Comment:
nit: avoid using the word 'spawnable' since it can be mixed up with actual '.spawnable' assets.